responses/_recorder.py

```diff
@@ -0,0 +1,98 @@
+from functools import wraps
+from itertools import groupby
+from unittest import mock as std_mock
+from urllib.parse import parse_qsl
+from urllib.parse import urlsplit
+
+from responses import Response
+from responses import _real_send
+from responses.registries import OrderedRegistry
+
+
+class Recorder:
```

Comment:
There is a good amount of duplication between the existing responses activation code and this. Should we first focus on loading 'recordings' and not on creating the 'man in the middle' to generate recordings?

I see the recording loader as the more valuable piece as it lets you refactor existing tests to share setup code or response fixtures.

Reply:
that is the problem. I think first we need to decide how to write the data, since this might be more tricky. Then we can read it back

since if we go in opposite direction, we may create an incompatible interface


Activation and some parts could be refactored, for sure.
Now we need to validate the idea and the best output format

Reply:
Isn't the data format dictated by what we need to load though? If we can't "record" all of that data we could record as much as possible. One wrinkle I see with recording is how custom registries work. I don't think we want to persist the registry state into the response file.

Would a minimal solution be to focus on loading a list of response definitions (without worrying about function responses) and then call `responses.add()` for each item in the bundle of responses?